src/app/dashboard-layout/courses-list/search-form/search-form.component.ts

```diff
@@ -16,22 +24,46 @@ const { faSearch } = FA_ICONS;
   styleUrls: ["./search-form.component.scss"],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SearchFormComponent {
+export class SearchFormComponent implements OnDestroy {
+  private destroySubj: Subject<boolean> = new Subject();
+  private inputValueSubj: Subject<string> = new Subject();
+
   /**
    * Fontawesome icons
    */
   public faSearch: IconDefinition = faSearch;
 
   /**
-   * Emit search value
+   * Observable of users' input value
    */
   @Output()
-  public search: EventEmitter<string> = new EventEmitter();
+  public search$: Observable<string>;
+
+  constructor() {
+    // Using an operator like takeUntil instead
+    // of manually unsubscribing will also complete
+    // the observable, triggering any completion event
+    // on the observable.
+    this.search$ = this.inputValueSubj.asObservable().pipe(
+      distinctUntilChanged((prevValue, currValue) => prevValue.trim() === currValue.trim()),
+      filter(value => !value.length || value.trim().length >= MIN_CHARS),
+      debounceTime(DEBOUNCE_TIME),
+      takeUntil(this.destroySubj),
+    );
+  }
+
+  /**
+   * ngOnDestroy
+   */
+  public ngOnDestroy(): void {
```

Comment:
Parent will unsubscribe onDestroy, so there is no need in destroy implementation here as far as you are not subscribing in the component